torch/_ops.py

```diff
@@ -26,9 +32,127 @@ def dl_open_guard():
         sys.setdlopenflags(old_flags)
 
 
+# TODO(voz) We are missing an entire axis of registration - Modes for the python key
+class PyOperatorABC(ABC):
+    def __call__(self, *args, **kwargs):
+        pass
+
+    def py_impl(self, dispatch_key, fn):
+        pass
+
+    def name(self):
+        pass
+
+    def dispatch(self, dispatch_key, *args, **kwargs):
+        pass
+
+
+class PyOperator(PyOperatorABC):
+    def __init__(self, name):
+        self._name = name
+        self.table = {}
+        self.python_key_mode_table = {}
+
+        # Make _OPNamespace not scream, this whole name based association needs a good hard look
+        self.__name__ = "pyop." + name
+        pyop_namespace.py_ops[name] = self
+
+    def fallthrough(self, dispatch_key):
+        self.table[dispatch_key] = self._fallthrough_fn(self, dispatch_key)
+
+    def py_impl(self, dispatch_key_or_mode):
+        def inner(fn):
+            if isinstance(dispatch_key_or_mode, type(TorchDispatchMode)):
```

Comment:
don't you want issubclassof

Reply:
Sure, that's cleaner

Reply:
It's not just cleaner, the current code is wrong lol (in the absence of metaclass the type will always be type)